util/reggen/ip_block.py

```diff
@@ -388,7 +366,7 @@ def from_raw(param_defaults: List[Tuple[str, str]],
 
         # Check that register blocks are in bijection with device interfaces
         reg_block_names = reg_blocks.keys()
-        dev_if_names = []  # type: List[Optional[str]]
+        dev_if_names = []  # type: list[str | None]
```

Comment:
You can annotate a type like this right?

```suggestion
        dev_if_names: list[str | None] = []
```